smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -128,20 +131,67 @@ contract MixinTransfer is
     public
   {
     _isValidKey(_tokenId);
-    if(
-      // the specified address does not own the token
-      ownerOf(_tokenId) != _from
-      || 
-      ( // the sender is not owner or authorized
-        ownerOf(_tokenId) != msg.sender
-        && approved[_tokenId] != msg.sender
-        && !isApprovedForAll(_ownerOf[_tokenId], msg.sender)
-      )
-      || // a key manager is set
-      keyManagerOf[_tokenId] != address(0)
-    ) {
+    _onlyKeyManagerOrApproved(_tokenId);
+    
+    // revert if key manager is set
+    if(keyManagerOf[_tokenId] != address(0)) {
```

Comment:
I think we don't need to revert, but just reset the keyManager to be 0x0

Reply:
yep makes sense, will just reset the key manager on transfer then